builds/e2e/templates/nested-parent-vm-setup.yaml

```diff
@@ -2,7 +2,25 @@ parameters:
   upstream.protocol: 'amqp'
 
 jobs:
+  - job: Lock_Nested_Agents_${{ parameters['upstream.protocol'] }}
+    displayName: Lock nested agents for ${{ parameters['upstream.protocol'] }}
+    timeoutInMinutes: 180
+    pool:
+     name: $(pool.name)
+     demands:
+       - agent-group -equals $(agent.group)
+       - Agent.OS -equals Linux
+       - Agent.OSArchitecture -equals X64
+       - status -equals unlocked
+    steps:
+    - template: nested-get-secrets.yaml
+    - script: scripts/linux/nestedAgentLock.sh -p $(edgebuild-1-PAT) -a $(agent.group) -b $(Build.BuildId)
```

Comment:
The `$(edgebuild-1-PAT)` along with other variables will be evaluated and use that as a command line. This mean the PAT will be stored in the test agent log. Please set it as
```
    - script: scripts/linux/nestedAgentLock.sh -p "$PAT" -a "$(agent.group)" -b "$(Build.BuildId)"
      env:
        PAT: $(edgebuild-1-PAT)
      displayName: Lock agents for nested topology
      name: lock_test_agent
```

Reply:
Can you reference environment variables when calling a script like that? That is useful.

Reply:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables